### PR TITLE
feat(discover2): Sort the list of fields, tags, and aggregate functions in alphabetical order

### PR DESCRIPTION
**Pull request by dashed**
Kapture 2019-11-25 at 17 09 42

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx [hunk 1]
```diff
@@ -8,13 +8,17 @@ export type ColumnValueType =
   | 'boolean'
   | 'never'; // Matches to nothing
 
+type AGGREGATION_KEYS = 'count' | 'count_unique' | 'min' | 'max' | 'sum' | 'avg';
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx]: Could this type be generated using `keyof AGGREGATIONS` after `AGGREGATIONS` is defined? Or use an untyped map to generate the typed one? Right now adding a new function requires updates in multiple places.

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx]: @markstory the alternative I can think of is this https://github.com/getsentry/sentry/pull/15819/commits/3bc8fa248de829291a20c9e8ac8cc9d42756e8b0

File: src/sentry/static/sentry/app/types/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// from:
+// - https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions
+// - https://www.typescriptlang.org/play/#example/assertion-functions
+
+// This declares a function which asserts that the expression called
+// value is true:
+// eslint-disable-next-line prettier/prettier
+export function assert(_value: unknown): asserts _value {}
```
Comment (markstory) [inline src/sentry/static/sentry/app/types/utils.tsx]: Shouldn't this have an assertion inside if the value isn't truthy?

Comment (dashed) [reply src/sentry/static/sentry/app/types/utils.tsx]: Nah. It shouldn't. It's not a runtime check.

Comment (dashed) [reply src/sentry/static/sentry/app/types/utils.tsx]: To clarify, this is a type assertion check. 

It's not to be confused with other utility functions like `invariant`, which is a runtime check: https://github.com/zertosh/invariant

**Pull request merged**